src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -440,7 +460,20 @@ public static JsonTypeInfo CreateJsonTypeInfo(Type type, JsonSerializerOptions o
         /// <returns>JsonPropertyInfo instance</returns>
         public JsonPropertyInfo CreateJsonPropertyInfo(Type propertyType, string name)
         {
-            ValidateType(propertyType, Type, null, Options);
+            if (propertyType == null)
+            {
+                throw new ArgumentNullException(nameof(propertyType));
+            }
+
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (IsInvalidForSerialization(propertyType))
+            {
+                ThrowHelper.ThrowArgumentException_CannotSerializeInvalidType(nameof(propertyType), propertyType, Type, name);
+            }
```

Comment:
Does the method check if `TypeInfoKind == JsonTypeInfoKind.Object`. We migth want to throw `InvalidOperationException` if we call the create method for an invalid declaring type.

Reply:
I thought we discussed this and decided not to throw. We won't let you add such property but no problem with creating

Reply:
Fair enough -- probably not too important.